signer-npm/src/lib.rs

```diff
@@ -431,6 +429,131 @@ pub fn cancel_multisig(
     Ok(multisig_transaction_js)
 }
 
+#[wasm_bindgen(js_name = createPymtChan)]
+pub fn create_pymtchan(
+    from_address: String,
+    to_address: String,
+    amount: String,
+    nonce: u32,
+) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+
+    let pch_transaction =
+        filecoin_signer::create_pymtchan(from_address, to_address, amount, nonce as u64).map_err(
+            |e| JsValue::from_str(format!("Error creating payment channel: {}", e).as_str()),
+        )?;
+
+    let pch_transaction_js = JsValue::from_serde(&pch_transaction)
+        .map_err(|e| JsValue::from(format!("Error creating transaction: {}", e)))?;
+
+    Ok(pch_transaction_js)
+}
+
+#[wasm_bindgen(js_name = settlePymtChan)]
+pub fn settle_pymtchan(
+    pch_address: String,
+    from_address: String,
+    nonce: u32,
+) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+
+    let pch_transaction = filecoin_signer::settle_pymtchan(pch_address, from_address, nonce as u64)
+        .map_err(|e| {
+            JsValue::from_str(format!("Error collecting payment channel: {}", e).as_str())
+        })?;
+
+    let pch_transaction_js = JsValue::from_serde(&pch_transaction)
+        .map_err(|e| JsValue::from(format!("Error creating transaction: {}", e)))?;
+
+    Ok(pch_transaction_js)
+}
+
+#[wasm_bindgen(js_name = collectPymtChan)]
+pub fn collect_pymtchan(
+    pch_address: String,
+    from_address: String,
+    nonce: u32,
+) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+
+    let pch_transaction =
+        filecoin_signer::collect_pymtchan(pch_address, from_address, nonce as u64).map_err(
+            |e| JsValue::from_str(format!("Error collecting payment channel: {}", e).as_str()),
+        )?;
+
+    let pch_transaction_js = JsValue::from_serde(&pch_transaction)
+        .map_err(|e| JsValue::from(format!("Error creating transaction: {}", e)))?;
+
+    Ok(pch_transaction_js)
+}
+
+#[wasm_bindgen(js_name = updatePymtChan)]
+pub fn update_pymtchan(
+    pch_address: String,
+    from_address: String,
+    signed_voucher: String,
+    nonce: u32,
+) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+
+    // TODO: verify if `pch_address` is an actor address. Not needed but good improvement.
```

Comment:
We have a problem with the amount of time this takes..
Maybe it is a good idea to create an issue for this improvement